corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -25,7 +24,13 @@ def JSON(obj):
     # json.dumps does not properly convert QueryDict array parameter to json
     if isinstance(obj, QueryDict):
         obj = dict(obj)
-    return mark_safe(json.dumps(obj, default=json_handler))
+    try:
+        return mark_safe(json.dumps(obj, default=json_handler))
+    except TypeError as e:
+        msg = ("Unserializable data was sent to the `|JSON` template tag.  "
+               "If DEBUG is off, Django will silently swallow this error.")
```

Comment:
is it possible to log more information about this error? this actually seems like a great use case for `soft_assert` which will give us a stack trace and also allow us to add a backoff in case this happens a lot. I would also try to include `obj` in the logging.